include/Animal.h

```diff
@@ -0,0 +1,33 @@
+#ifndef ANIMAL_H
+#define ANIMAL_H
+
+#include <iostream>
+#include "Location.h"
+
+class Animal
+{
+public:
+	Animal(const std::string& name,const Location& location);
+	~Animal();
```

Comment:
no use of virtual destructor 
check this
https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors

Reply:
fixed